general 32-bit (i386) support

A user reported build failures on [32-bit FreeBSD](https://pkg-status.freebsd.org/beefy15/data/132i386-default/8b3f09304cc8/logs/h2o-devel-2.3.0.d.20230817.log)
 
```
error: call to undeclared function '_mm_insert_epi64'; ISO C99 and later do not support implicit function declarations [-Werror,-Wimplicit-function-declaration]
    ctr = _mm_insert_epi64(ctr, seq, 0);
          ^
/wrkdirs/usr/ports/www/h2o-devel/work/h2o-70dd2d8/deps/picotls/lib/fusion.c:1129:11: note: did you mean '_mm_set_epi64'?
/usr/lib/clang/16/include/emmintrin.h:3447:46: note: '_mm_set_epi64' declared here
static __inline__ __m128i __DEFAULT_FN_ATTRS _mm_set_epi64(__m64 __q1,
                                             ^
/wrkdirs/usr/ports/www/h2o-devel/work/h2o-70dd2d8/deps/picotls/lib/fusion.c:1129:9: error: assigning to '__m128i' (vector of 2 'long long' values) from incompatible type 'int'
    ctr = _mm_insert_epi64(ctr, seq, 0);
        ^ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
2 errors generated.
```

Does h2o support 32-bit platforms or should I just skip this architecture from now on?

--- a/deps/picotls/cmake/fusion.cmake
+++ b/deps/picotls/cmake/fusion.cmake
@@ -14,6 +14,7 @@ FUNCTION (CHECK_FUSION_PREREQUISITES)
         ord0 = _mm256_aesenc_epi128(ord1, ord2);
         ord3 = _mm256_aesenclast_epi128(ord0, ord1);
         ord1 = _mm256_clmulepi64_epi128(ord3, ord2, 0x00);
+        _mm_insert_epi64(_mm_setr_epi32(0, 1, 2, 3), 0, 0);
         return 0;
     }
     " CC_HAS_AESNI256)

